sdk/src/stream_config/config_iggy_consumer.rs

```diff
@@ -0,0 +1,369 @@
+    batch_size: u32,
+    create_stream_if_not_exists: bool,
+    create_topic_if_not_exists: bool,
+    consumer_name: String,
+    consumer_kind: ConsumerKind,
+    polling_interval: IggyDuration,
+    polling_strategy: PollingStrategy,
+    partitions_count: u32,
+    replication_factor: Option<u8>,
+    encryptor: Option<Arc<EncryptorKind>>,
+}
+
+impl Default for IggyConsumerConfig {
+    fn default() -> Self {
+        let stream_id = shared_config::get_identifier_from_string("test_stream");
+        let topic_id = shared_config::get_identifier_from_string("test_topic");
+
+        Self {
+            stream_id,
+            stream_name: "test_stream".to_string(),
+            topic_id,
+            topic_name: "test_topic".to_string(),
+            auto_commit: AutoCommit::When(AutoCommitWhen::PollingMessages),
+            batch_size: 100,
+            create_stream_if_not_exists: false,
+            create_topic_if_not_exists: false,
+            consumer_name: "test_consumer".to_string(),
+            consumer_kind: ConsumerKind::ConsumerGroup,
+            polling_interval: IggyDuration::from_str("5ms").unwrap(),
+            polling_strategy: PollingStrategy::last(),
+            partitions_count: 1,
+            replication_factor: None,
+            encryptor: None,
+        }
+    }
+}
+
+impl IggyConsumerConfig {
+    /// Creates a new `IggyConsumerConfig` from the given arguments.
+    ///
+    /// # Args
+    ///
+    /// * `stream_id` - The stream id.
+    /// * `stream_name` - The stream name.
+    /// * `topic_id` - The topic id.
+    /// * `topic_name` - The topic name.
+    /// * `auto_commit` - The auto commit config.
+    /// * `batch_size` - The max number of messages to send in a batch.
+    /// * `create_stream_if_not_exists` - Whether to create the stream if it does not exists.
+    /// * `create_topic_if_not_exists` - Whether to create the topic if it does not exists.
+    /// * `consumer_name` - The consumer name.
+    /// * `consumer_kind` - The consumer kind.
+    /// * `polling_interval` - The interval between polling for new messages.
+    /// * `polling_strategy` - The polling strategy.
+    /// * `partitions_count` - The number of partitions.
+    /// * `replication_factor` - The replication factor.
+    /// * `encryptor` - The encryptor.
+    ///
+    /// Returns:
+    /// A new `IggyConsumerConfig`.
+    ///
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        stream_id: Identifier,
+        stream_name: String,
+        topic_id: Identifier,
+        topic_name: String,
+        auto_commit: AutoCommit,
+        batch_size: u32,
+        create_stream_if_not_exists: bool,
+        create_topic_if_not_exists: bool,
+        consumer_name: String,
+        consumer_kind: ConsumerKind,
+        polling_interval: IggyDuration,
+        polling_strategy: PollingStrategy,
+        partitions_count: u32,
+        replication_factor: Option<u8>,
+        encryptor: Option<Arc<EncryptorKind>>,
```

Comment:
As mentioned previously, let's consider extending this with `init_retries` (retry amount + interval).